### PR TITLE
Adding spatial index

### PR DESCRIPTION
**Pull request by ferdonline**
Adding the initial Spatial Index v0.0.1

### CODE REVIEW

File: var/spack/repos/builtin/packages/spatial-index/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    """Spatial indexer for geometries and morphologies"""
+
+    homepage = "http://www.example.com"
+    git      = "ssh://bbpcode.epfl.ch/hpc/SpatialIndex"
+
+    depends_on("py-setuptools")
+    depends_on("cmake")
+    depends_on("boost")
+
+    version('0.0.1', git=git, tag='0.0.1', submodules=True)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/spatial-index/package.py]:
```suggestion
    version('0.0.1', tag='0.0.1', submodules=True)
```

File: var/spack/repos/builtin/packages/spatial-index/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack import *
+
+
+class SpatialIndex(PythonPackage):
+    """Spatial indexer for geometries and morphologies"""
+
+    homepage = "http://www.example.com"
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/spatial-index/package.py]:
```suggestion
    homepage = "https://bbpcode.epfl.ch/code/#/admin/projects/hpc/SpatialIndex"
```

File: var/spack/repos/builtin/packages/spatial-index/package.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    git      = "ssh://bbpcode.epfl.ch/hpc/SpatialIndex"
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/spatial-index/package.py:12]:
```suggestion
    git      = "ssh://bbpcode.epfl.ch/hpc/SpatialIndex"
    url      = "ssh://bbpcode.epfl.ch/hpc/SpatialIndex"
```

Makes it easier to use `diy` with fantasy versions.

**Review by matz-e — CHANGES_REQUESTED** [review]

**Review by ferdonline — COMMENTED** [review]
Fine w me

**Review by matz-e — APPROVED** [review]

**Comment by ferdonline** [comment]
@matz-e Is it a good idea to use the same PR to enable the module and deployment?

**Comment by matz-e** [comment]
Yes, IMO.

File: var/spack/repos/builtin/packages/spatial-index/package.py [hunk 4]
```diff
@@ -16,6 +16,6 @@ class SpatialIndex(PythonPackage):
     depends_on("cmake")
     depends_on("boost")
 
-    version('0.0.1', tag='0.0.1', submodules=True)
+    version('0.0.1', tag='0.0.1-2', submodules=True)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/spatial-index/package.py:19]: Really? We can't just call that `0.0.2`? It's just a patch release…

Comment (ferdonline) [reply var/spack/repos/builtin/packages/spatial-index/package.py:19]: It's still a deployed version, so we'd have to re deploy documentation with a new version, etc... This way it's obvious we're doing a quick fix to 0.0.1

**Pull request merged**